core/embed/sec/optiga/optiga.c

```diff
@@ -259,13 +272,45 @@ uint32_t optiga_estimate_time_ms(storage_pin_op_t op) {
   // To estimate the overall time of the PIN operation we multiply the
   // throttling delay by the number of protected Optiga commands and add the
   // time required to execute all Optiga commands without throttling delays.
+
+  const int init_metadata_time = 550;
```

Comment:
I'd expect `init_metadata_time` to depend on `STRETCHED_PIN_COUNT`. Something like `(7 + STRETCHED_PIN_COUNT) * set_metadata_time`.